lib/FrontendTool/FrontendTool.cpp

```diff
@@ -958,6 +958,20 @@ static bool performCompile(CompilerInstance &Instance,
   return false;
 }
 
+static Optional<bool> emitSILAfterSILGen(CompilerInvocation &Invocation,
```

Comment:
Similarly, besides the Action test, this is the same work we do to emit SIL at *any* stage…except the `performSILLinking` part.

Reply:
Refactored differently, into linkAllIfNeeded, and writeSIL with fewer arguments. Looks nice and concise now, with common code deduplicated.